zenith_utils/src/pq_proto.rs

```diff
@@ -0,0 +1,559 @@
+        // uses more than one prepared statement at a time.
+        /*
+        if !pstmt_name.is_empty() {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "named prepared statements not implemented in Parse",
+            ));
+        }
+         */
+
+        if nparams != 0 {
+            bail!("query params not implemented");
+        }
+
+        Ok(FeMessage::Parse(FeParseMessage { query_string }))
+    }
+}
+
+impl FeDescribeMessage {
+    pub fn parse(mut buf: Bytes) -> anyhow::Result<FeMessage> {
+        let kind = buf.get_u8();
+        let _pstmt_name = read_null_terminated(&mut buf)?;
+
+        // FIXME: see FeParseMessage::parse
+        /*
+        if !pstmt_name.is_empty() {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "named prepared statements not implemented in Describe",
+            ));
+        }
+        */
+
+        if kind != b'S' {
+            bail!("only prepared statmement Describe is implemented");
+        }
+
+        Ok(FeMessage::Describe(FeDescribeMessage { kind }))
+    }
+}
+
+impl FeExecuteMessage {
+    pub fn parse(mut buf: Bytes) -> anyhow::Result<FeMessage> {
+        let portal_name = read_null_terminated(&mut buf)?;
+        let maxrows = buf.get_i32();
+
+        if !portal_name.is_empty() {
+            bail!("named portals not implemented");
+        }
+
+        if maxrows != 0 {
+            bail!("row limit in Execute message not supported");
+        }
+
+        Ok(FeMessage::Execute(FeExecuteMessage { maxrows }))
+    }
+}
+
+impl FeBindMessage {
+    pub fn parse(mut buf: Bytes) -> anyhow::Result<FeMessage> {
+        let portal_name = read_null_terminated(&mut buf)?;
+        let _pstmt_name = read_null_terminated(&mut buf)?;
+
+        if !portal_name.is_empty() {
+            bail!("named portals not implemented");
+        }
+
+        // FIXME: see FeParseMessage::parse
+        /*
+        if !pstmt_name.is_empty() {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "named prepared statements not implemented",
+            ));
+        }
+        */
+
+        Ok(FeMessage::Bind(FeBindMessage {}))
+    }
+}
+
+impl FeCloseMessage {
+    pub fn parse(mut buf: Bytes) -> anyhow::Result<FeMessage> {
+        let _kind = buf.get_u8();
+        let _pstmt_or_portal_name = read_null_terminated(&mut buf)?;
+
+        // FIXME: we do nothing with Close
+
+        Ok(FeMessage::Close(FeCloseMessage {}))
+    }
+}
+
+fn read_null_terminated(buf: &mut Bytes) -> anyhow::Result<Bytes> {
+    let mut result = BytesMut::new();
+
+    loop {
+        if !buf.has_remaining() {
+            bail!("no null-terminator in string");
+        }
+
+        let byte = buf.get_u8();
+
+        if byte == 0 {
+            break;
+        }
+        result.put_u8(byte);
+    }
+    Ok(result.freeze())
```

Comment:
Not your code, but perhaps we could simplify this by `.splitn(2, 0u8)`

Reply:
It would hardly get shorter, as `split` returns two slices while here we move data from one Bytes to the new one...

Reply:
Ahh, maybe we could use the method `split_off` on Bytes